corehq/apps/es/tests/test_filters.py

```diff
@@ -201,26 +201,38 @@ def test_source_include(self):
         self.checkQuery(q, json_output)
 
 
-class TestNotEdgeCase(SimpleTestCase):
+@es_test
+class TestFiltersRun(SimpleTestCase):
     def setUp(self):
         self.es = get_es_new()
         self.index = XFORM_INDEX_INFO.index
 
     def tearDown(self):
         ensure_index_deleted(self.index)
 
-    def test_assume_alias(self):
+    def _setup_data(self):
         initialize_index_and_mapping(self.es, XFORM_INDEX_INFO)
         doc1 = {'_id': 'doc1', 'domain': 'd', 'app_id': 'a'}
         doc2 = {'_id': 'doc2', 'domain': 'd', 'app_id': 'not_a'}
         doc3 = {'_id': 'doc3', 'domain': 'not_d', 'app_id': 'not_a'}
         for doc in [doc1, doc2, doc3]:
             send_to_elasticsearch('forms', doc)
         self.es.indices.refresh(self.index)
+
+    def test_not_filter_edge_case(self):
+        self._setup_data()
         query = FormES().remove_default_filters().filter(
             filters.NOT(filters.OR(
                 filters.term('domain', 'd'),
                 filters.term('app_id', 'a')
             ))
         )
         self.assertEqual(query.run().doc_ids, ['doc3'])
+
+    def test_ids_query(self):
+        self._setup_data()
+        ids = ['doc1', 'doc2']
+        self.assertEqual(
+            FormES().remove_default_filters().ids_query(ids).exclude_source().run().doc_ids,
```

Comment:
is it necessary to call `remove_default_filters`

Reply:
Yes, because `_setup_data` doesn't setup a doc that looks exactly like xform and the default filters look for docs that are xforms.

Reply:
If you're looking for IDs don't you think it should remove any other filters automatically?

Reply:
It's expected that all the real data will pass through the default filters. In this case, since it's just tests, I haven't setup realistic data, so I need to remove those filters.